scala/private/rule_impls.bzl

```diff
@@ -827,6 +833,50 @@ def _scala_binary_common(
         runfiles = runfiles,
     )
 
+def _pack_source_jars(ctx):
+  source_jars = []
+
+  # collect .java sources and pack a source jar for Java
+  java_sources = [
+      f for f in ctx.files.srcs if f.basename.endswith(_java_extension)
+  ]
+  if len(java_sources) > 0:
+      pseudo_java_jar_file = ctx.actions.declare_file(ctx.label.name + "_java.jar")
+      java_source_jar = java_common.pack_sources(
```

Comment:
Hmm, maybe we actually get this already from the java_common.compile?

Reply:
This is not working yet BTW. Meaning, the java sources are not auto-attached like the Scala sources do. I think it's because of the jar root dir issue, but I haven't proved that yet.
What do you mean by get it from the java_common.compile? The Java sources jar? I don't think so. It does generate a separate `_java.jar` file, but not the sources. Do you mean that we can get that by configuring it differently? Does it have a flag for source gen or something?